### PR TITLE
fix(module-tools): alias not working on re-export

### PR DESCRIPTION
**Pull request by 10Derozan**
### <samp>🤖 Generated by Copilot at ab6b6ad</samp>

This pull request fixes a bug in the `redirect` feature of the `@modern-js/module-tools` package, which did not work on re-export statements with aliases. It also adds test cases and updates the changeset file for the patch version update.

### <samp>🤖 Generated by Copilot at ab6b6ad</samp>

*  Update changeset file to indicate patch version update and fix messages for `@modern-js/module-tools` ([link](https://github.com/web-infra-dev/modern.js/pull/4784/files?diff=unified&w=0#diff-696716c82edf7cfcaca7bb7631b51c49224c13169700917a9c19f0764178cb4fR1-R6))
*  Add two more cases to `redirect` feature to handle double quotes in export statements ([link](https://github.com/web-infra-dev/modern.js/pull/4784/files?diff=unified&w=0#diff-d664fc1d40136750c5a0c4a3a5aa1a225ddca0288937f8d80c60d12d215b3ba0R210-R211))
*  Add test fixture files `alias.ts` and `index.ts` to demonstrate alias rewriting by `redirect` feature ([link](https://github.com/web-infra-dev/modern.js/pull/4784/files?diff=unified&w=0#diff-b8960cb51a48abc91342c04fc2a92b78e168de615ce51846a019e6233ce3a830R3), [link](https://github.com/web-infra-dev/modern.js/pull/4784/files?diff=unified&w=0#diff-cae32e7ec5ad25c770b054a3e596045126827d7275498bb592581ac28ca4488dR5))
*  Add assertion to integration test for `redirect` feature to check output file does not include original alias ([link](https://github.com/web-infra-dev/modern.js/pull/4784/files?diff=unified&w=0#diff-6120d7373188e5bb1bb4404d8c7fc3f12cef9eaa21cdc8bbc158f5139c8081efR34))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/solutions/module-tools/src/builder/feature/redirect.ts [hunk 1]
```diff
@@ -207,6 +207,8 @@ export const redirect = {
         const staticPattern = [
           `import $$VAR from '$MATCH'`,
           `import $$VAR from "$MATCH"`,
+          `export {$$VAR} from '$MATCH'`,
+          `export {$$VAR} from "$MATCH"`,
```
Comment (chenjiahan) [inline packages/solutions/module-tools/src/builder/feature/redirect.ts:211]: how about `import * as $$VAR from` and `export * from`?

Comment (chenjiahan) [reply packages/solutions/module-tools/src/builder/feature/redirect.ts:211]: And there may be some other cases, such as `import type`

Comment (chenjiahan) [reply packages/solutions/module-tools/src/builder/feature/redirect.ts:211]: 👍 Can we add some test cases for `import * as` and `export * from`

Comment (10Derozan) [reply packages/solutions/module-tools/src/builder/feature/redirect.ts:211]: sure, i'll do it. i have add case about `export * from`

**Pull request merged**